docs/content/quickstart/_index.md

```diff
@@ -128,6 +129,27 @@ Upgraded to World 2.0
 execution completed successfully!
 ```
 
+### Upgrade using digest
+
+When working in production environment, we highly recommend you to reference the bundle using its digest instead of tag. We used tags in our docs for simplicity, but tags can be overwritten which results in unexpected outcomes from upgrading an bundle.
+To get deterministic and repeatable deployments, use digests instead of tags. it allows to ensure that you deploy the exact expected bundle.
+
+```console
+$ porter upgrade porter-hello --reference ghcr.io/getporter/examples/porter-hello@sha256:276b44be3f478b4c8d1f99c1925386d45a878a853f22436ece5589f32e9df384
+_
+upgrading porter-hello...
+executing upgrade action from porter-hello (installation: porter-hello)
+Upgrade Hello World
+Upgraded to World 2.0
+execution completed successfully!
+```
+
+If you ran into below error message:
+```console
+$ unable to pull bundle: invalid media type "application/vnd.docker.distribution.manifest.v2+json" for bundle manifest
+```
+Make sure that you are referencing the digest for bundles and not the digest for invocation images. 
+
```

Comment:
Let's explicitly call this out as a troubleshooting section, so that we can include other potential errors or mistakes in the future.

```suggestion
## Troubleshooting

If you receive an `invalid media type` error like below, check that you are referencing the digest for the bundle and not the installer image.

```plain
unable to pull bundle: invalid media type "application/vnd.docker.distribution.manifest.v2+json" for bundle manifest
```

```